iceoryx_hoofs/vocabulary/include/iox/string.hpp

```diff
@@ -423,6 +424,13 @@ class string final
     template <typename T>
     IsStringOrCharArrayOrChar<T, bool> unsafe_append(const T& str) noexcept;
 
+    /// @brief direct access to the string's raw pointer. The access resizes the data correctly after the function call.
+    /// If the data written in the function exceeds the capacity, and the code does not segfault beforehand due to an
+    /// invalid access, a FATAL error occurs.
+    ///
+    /// @param [in] function composed by the raw data, current string size and the capacity
```

Comment:
nitpick
```suggestion
    /// @param [in] func is a function composed by the raw data pointer, current string size and the capacity
```

Could you also add a small example how this is used, nothing fancy just something like this
```cpp
iox::string<100> s;
s.unsafe_raw_access([] (auto* str, const uint64_t size, const uint64_t capacity) {
    strncpy(str, "Hello World", capacity);
});
```

Maybe also mention that is is safe to write the null-termination at `str[capacity]`.